### PR TITLE
Make 'recent-blog-entries' component consult the BlogPageCategory object set

### PR DESCRIPTION
**Pull request by Pomax**
Finally closes https://github.com/mozilla/foundation.mozilla.org/issues/3851

### CODE REVIEW

File: network-api/networkapi/wagtailpages/pagemodels/customblocks/recent_blog_entries.py [hunk 1]
```diff
@@ -72,10 +66,14 @@ def get_context(self, value, parent_context=None):
         if category:
             type = "category"
             query = slugify(category)
-            BlogPageCategory = apps.get_model('wagtailpages.BlogPageCategory')
-            category_object = BlogPageCategory.objects.get(name=category)
-            blogpage.extract_category_information(category_object.slug)
-            entries = blogpage.get_entries(context)
+            try:
+                # verify this category exists
+                category_object = BlogPageCategory.objects.get(name=category)
```
Comment (mmmavis) [inline network-api/networkapi/wagtailpages/pagemodels/customblocks/recent_blog_entries.py:71]: This doesn't seem to work when category is `all`. It goes to `except` and hence does not return after result back.

File: network-api/networkapi/wagtailpages/pagemodels/customblocks/recent_blog_entries.py [hunk 2]
```diff
@@ -72,10 +66,14 @@ def get_context(self, value, parent_context=None):
         if category:
             type = "category"
             query = slugify(category)
-            BlogPageCategory = apps.get_model('wagtailpages.BlogPageCategory')
-            category_object = BlogPageCategory.objects.get(name=category)
-            blogpage.extract_category_information(category_object.slug)
-            entries = blogpage.get_entries(context)
+            try:
```
Comment (mmmavis) [inline network-api/networkapi/wagtailpages/pagemodels/customblocks/recent_blog_entries.py:69]: I can't recall how `tag` is set up. Curious why only `category` has to go through try-except and `tag` deosn't?

File: network-api/networkapi/wagtailpages/pagemodels/customblocks/recent_blog_entries.py (hunk 1 above)
Comment (Pomax) [reply network-api/networkapi/wagtailpages/pagemodels/customblocks/recent_blog_entries.py:71]: excellent point!

File: network-api/networkapi/wagtailpages/pagemodels/customblocks/recent_blog_entries.py (hunk 2 above)
Comment (Pomax) [reply network-api/networkapi/wagtailpages/pagemodels/customblocks/recent_blog_entries.py:69]: tag gets try/excepted over in `filter_entries_for_tag`, called as part of `get_entries`

Comment (Pomax) [reply network-api/networkapi/wagtailpages/pagemodels/customblocks/recent_blog_entries.py:69]: https://github.com/mozilla/foundation.mozilla.org/blob/master/network-api/networkapi/wagtailpages/pagemodels/index.py#L113-L120

**Pull request merged**